src/main/java/com/targus/ui/WSNPrototype.java

```diff
@@ -0,0 +1,196 @@
+package com.targus.ui;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.targus.utils.Constants;
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.geometry.Point2D;
+import javafx.stage.FileChooser;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class WSNPrototype {
+    private int paneWidth;
+    private int paneHeight;
+    private final ArrayList<Point2D> potentialPositions = new ArrayList<Point2D>();
+
+    public ArrayList<Point2D> getPotentialPositions() {
+        return potentialPositions;
+    }
+
+    private final ArrayList<Point2D> targets = new ArrayList<Point2D>();
+
+    public ArrayList<Point2D> getTargets() {
+        return targets;
+    }
+
+    private final IntegerProperty mProperty = new SimpleIntegerProperty(1);
+
+    public IntegerProperty getMProperty() {
+        return mProperty;
+    }
+
+    private final IntegerProperty kProperty = new SimpleIntegerProperty(1);
+
+    public IntegerProperty getKProperty() {
+        return kProperty;
+    }
+
+    private final DoubleProperty communicationRangeProperty = new SimpleDoubleProperty(100);
+
+    public DoubleProperty getCommunicationRangeProperty() {
+        return communicationRangeProperty;
+    }
+
+    private final DoubleProperty sensingRangeProperty = new SimpleDoubleProperty(50);
+
+    public DoubleProperty getSensingRangeProperty() {
+        return sensingRangeProperty;
+    }
+
+    private final IntegerProperty generationCountProperty = new SimpleIntegerProperty(1000);
+
+    public IntegerProperty getGenerationCountProperty() {
+        return generationCountProperty;
+    }
+
+    private final DoubleProperty mutationRateProperty = new SimpleDoubleProperty(0.3);
+
+    public DoubleProperty getMutationRateProperty() {
+        return mutationRateProperty;
+    }
+
+    public int getPaneWidth() {
+        return paneWidth;
+    }
+
+    public int getPaneHeight() {
+        return paneHeight;
+    }
+
+    public void setPaneWidth(int paneWidth) {
+        this.paneWidth = paneWidth;
+    }
+
+    public void setPaneHeight(int paneHeight) {
+        this.paneHeight = paneHeight;
+    }
+
+    public void exportToFile() {
+        try {
+            FileChooser fc = new FileChooser();
+            fc.setInitialDirectory(new File("."));
+            fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("JSON Files", "*.json"));
+            File f = fc.showSaveDialog(null);
+
+            if (f == null) {
+                return;
+            }
+
+            String src = f.getAbsolutePath();
+
+            BufferedWriter writer = Files.newBufferedWriter(Paths.get(src));
```

Comment:
We can extract this writing to JSON file operations into a separate private method and call it.

```java
private Map<String, Object> <method_name> () {}
```

Reply:
Adds to the complexity without a significant gain. So I thought it's better to keep it as is.